pam/return.go

```diff
@@ -1,54 +1,64 @@
 package main
 
+import (
+	"github.com/msteinert/pam"
+)
+
 // Various signalling return messaging to PAM.
 
-// pamSuccess signals PAM module to return PAM_SUCCESS and Quit tea.Model.
-type pamSuccess struct {
-	brokerID string
+// pamReturnStatus is the interface that all PAM return types should implement.
+type pamReturnStatus interface {
+	Message() string
 }
 
-// String returns the string of pamSuccess.
-func (err pamSuccess) String() string {
-	return ""
+// pamReturnError is an interface that PAM errors return types should implement.
+type pamReturnError interface {
+	pamReturnStatus
+	Status() pam.StatusError
 }
 
-// pamIgnore signals PAM module to return PAM_IGNORE and Quit tea.Model.
-type pamIgnore struct {
-	localBrokerID string // Only set for local broker to store it globally.
-	msg           string
+// pamSuccess signals PAM module to return with provided pam.Success and Quit tea.Model.
+type pamSuccess struct {
+	brokerID string
+	msg      string
 }
 
-// String returns the string of pamIgnore message.
-func (err pamIgnore) String() string {
-	return err.msg
+// Message returns the message that should be sent to pam as info message.
+func (p pamSuccess) Message() string {
+	return p.msg
 }
 
-// pamAbort signals PAM module to return PAM_ABORT and Quit tea.Model.
-type pamAbort struct {
-	msg string
+// pamIgnore signals PAM module to return pam.Ignore and Quit tea.Model.
+type pamIgnore struct {
+	msg           string
+	localBrokerID string // Only set for local broker to store it globally.
```

Comment:
Nitpick: can you put the msg after the localBrokerID? This way the struct structure is consistent among all return statuses.

Reply:
Oh sure I thought I did already for all types, but you're right.